### PR TITLE
tests: use assert in strict assertion mode

### PR DESCRIPTION
**Pull request by brendankenny**
[Strict assertion mode](https://nodejs.org/api/assert.html#assert_strict_assertion_mode) has been recommended since it was added in Node v9.9.0, and treats all the equality asserts as strict equality assertions. It also gives nicer deep equality diffs.

Not much was caught here, which is good news. A few `null` -> `undefined` and `number` -> `string` coercions, and a couple of old audit scores that weren't updated when all scores went numeric.

### CODE REVIEW

File: lighthouse-core/test/report/html/renderer/dom-test.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ describe('DOM', () => {
       const el = dom.createElement('div');
       assert.equal(el.localName, 'div');
       assert.equal(el.className, '');
-      assert.equal(el.className, el.attributes.length);
```
Comment (brendankenny) [inline lighthouse-core/test/report/html/renderer/dom-test.js:41]: no idea what this was trying to test, so I did my best :)

File: lighthouse-core/test/report/html/renderer/details-renderer-test.js [hunk 2]
```diff
@@ -414,8 +414,8 @@ describe('DetailsRenderer', () => {
       assert.equal(anchorEl.href, 'https://www.example.com/script.js');
       assert.equal(sourceLocationEl.textContent, '/script.js:11:5(www.example.com)');
       assert.equal(sourceLocationEl.getAttribute('data-source-url'), sourceLocation.url);
-      assert.equal(sourceLocationEl.getAttribute('data-source-line'), sourceLocation.line);
-      assert.equal(sourceLocationEl.getAttribute('data-source-column'), sourceLocation.column);
+      assert.equal(sourceLocationEl.getAttribute('data-source-line'), `${sourceLocation.line}`);
```
Comment (connorjclark) [inline lighthouse-core/test/report/html/renderer/details-renderer-test.js:417]: nit: you like this better than String()? was the line limit reached if you did that?

File: lighthouse-core/test/report/html/renderer/dom-test.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/test/report/html/renderer/dom-test.js:41]: lol this is hilarious

File: lighthouse-core/test/report/html/renderer/details-renderer-test.js (hunk 2 above)
Comment (brendankenny) [reply lighthouse-core/test/report/html/renderer/details-renderer-test.js:417]: > nit: you like this better than String()? was the line limit reached if you did that?

it does go over the limit on the `column` lines, but I also don't have a particular preference between the two

**Pull request merged**